codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -162,6 +171,7 @@ public void onHttpClientUpgrade() throws Http2Exception {
 
         // Create a local stream used for the HTTP cleartext upgrade.
         connection().local().createStream(HTTP_UPGRADE_STREAM_ID, true);
+        sendPreface();
```

Comment:
Client doesn't send preface in this way for Upgrade. Instead, the SETTINGS goes in the initial HTTP/1 Headers that contained the Upgrade request. This works because you can send redundant SETTINGS at any time, but is misleading and seems like it could simply be removed.

Reply:
true... So yeah I think we only need this for servers.

Reply:
Maybe I'm interpreting the spec differently, but I think clients do need to send the preface this way.

https://tools.ietf.org/html/rfc7540#section-3.2

```
   The first HTTP/2 frame sent by the server MUST be a server connection
   preface (Section 3.5) consisting of a SETTINGS frame (Section 6.5).
   Upon receiving the 101 response, the client MUST send a connection
   preface (Section 3.5), which includes a SETTINGS frame.
```

Reply:
@mosesn, you're right. That's weird given the `HTTP2-Settings` header but so be it.

Sorry @normanmaurer!